### PR TITLE
Ask for confirmation when leaving an edited form

### PR DESCRIPTION
**Pull request by danrot**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | ---
| Related issues/PRs | ---
| License | MIT
| Documentation PR | ---

#### What's in this PR?

This PR adds some hooks to interfere the routing process, in order to enable the form to avoid routing if it has been edited.

#### Why?

Because doing this leads to data loss, so the user should be asked for confirmation.

#### To Do

- [x] Create a documentation PR
- [x] Add breaking changes to UPGRADE.md

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/translations/admin.en.json [hunk 1]
```diff
@@ -103,5 +103,7 @@
     "sulu_admin.heading3": "Heading 3",
     "sulu_admin.heading4": "Heading 4",
     "sulu_admin.heading5": "Heading 5",
-    "sulu_admin.heading6": "Heading 6"
+    "sulu_admin.heading6": "Heading 6",
+    "sulu_admin.dirty_warning_dialog_title": "Attention!",
```
Comment (chirimoya) [inline src/Sulu/Bundle/AdminBundle/Resources/translations/admin.en.json]: Leave form?

File: src/Sulu/Bundle/AdminBundle/Resources/translations/admin.en.json [hunk 2]
```diff
@@ -103,5 +103,7 @@
...
+    "sulu_admin.dirty_warning_dialog_text": "There is unsaved data being lost when leaving."
```
Comment (chirimoya) [inline src/Sulu/Bundle/AdminBundle/Resources/translations/admin.en.json]: Changes you made will not be saved.

File: src/Sulu/Bundle/AdminBundle/Resources/translations/admin.de.json [hunk 3]
```diff
@@ -103,5 +103,7 @@
     "sulu_admin.heading3": "Überschrift 3",
     "sulu_admin.heading4": "Überschrift 4",
     "sulu_admin.heading5": "Überschrift 5",
-    "sulu_admin.heading6": "Überschrift 6"
+    "sulu_admin.heading6": "Überschrift 6",
+    "sulu_admin.dirty_warning_dialog_title": "Achtung!",
```
Comment (chirimoya) [inline src/Sulu/Bundle/AdminBundle/Resources/translations/admin.de.json]: Formular verlassen?

File: src/Sulu/Bundle/AdminBundle/Resources/translations/admin.de.json [hunk 4]
```diff
@@ -103,5 +103,7 @@
...
+    "sulu_admin.dirty_warning_dialog_text": "Es liegen nicht gespeicherte Daten vor, die beim Verlassen verloren gehen."
```
Comment (chirimoya) [inline src/Sulu/Bundle/AdminBundle/Resources/translations/admin.de.json]: Ihre Änderungen werden nicht gespeichert.

File: src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/README.md [hunk 5]
```diff
@@ -156,3 +156,20 @@ router.addUpdateAttributesHooks((route) => ({
...
 router.navigate('sulu_page.webspace_overview', {webspace: 'sulu_io', test: true})
 ```
+
+There are also the `updateRouteHooks`, which are called with the destination route, its attributes and a reference to
+the function that was called (`navigate`, `redirect` or `restore`). Each of these hooks can return `false`, which will
+result in the navigation not to happen. This way the application can e.g. stop the navigation process if some data is
+not saved yet (that's what it is used for in the [`Form`](#form) view).
+
+```javascript static
+router.addUpdateRouteHook((route) => {
```
Comment (chirimoya) [inline src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/README.md]: add all params

**Comment by danrot** [comment]
@chirimoya Had to introduce a `reload` method on the `ResourceStore` which is an action, because mobx strict mode was throwing an error when adding e.g. a new contact.

**Pull request merged**